xarray/tests/test_dataset.py

```diff
@@ -2719,6 +2719,12 @@ def test_where_drop_empty(self):
         expected = DataArray(np.zeros((0, 10)), dims=['nCells', 'nVertLevels'])
         self.assertDatasetIdentical(expected, actual)
 
+    def test_where_drop_no_indexes(self):
```

Comment:
Yes. I added some additional test coverage when we made indices optional but clearly not enough here.